ow_lander/scripts/activity_grind.py

```diff
@@ -7,52 +7,123 @@
 import constants
 import math
 import copy
-from tf.transformations import quaternion_from_euler
 from utils import is_shou_yaw_goal_in_range
-from activity_full_digging_traj import move_to_pre_trench_configuration
 from activity_full_digging_traj import go_to_Z_coordinate, change_joint_value
 
-def plan_cartesian_path_lin(move_arm, length):
+def arg_parsing(req):
+  if req.use_defaults :
+    # Default trenching values
+    x_start = 1.65
+    y_start = 0.0
+    depth = 0.05
+    length = 0.6
+    parallel = True
+    ground_position = constants.DEFAULT_GROUND_HEIGHT
+    delete_prev_traj = False
+
+  else :
+    x_start = req.x
+    y_start = req.y
+    depth = req.depth
+    length = req.length
+    parallel = req.parallel
+    ground_position = req.ground_position
+    delete_prev_traj = req.delete_prev_traj
+
+  return [req.use_defaults, x_start, y_start, depth, length, parallel, ground_position, delete_prev_traj]
+
+
+def plan_cartesian_path(move_group, length, alpha, parallel):
+
+  if parallel==False:
+    alpha = alpha - math.pi/2
 
   waypoints = []
-  wpose = move_arm.get_current_pose().pose
-  wpose.position.x += length
+  wpose = move_group.get_current_pose().pose
+  wpose.position.x += length*math.cos(alpha)
+  wpose.position.y += length*math.sin(alpha)
   waypoints.append(copy.deepcopy(wpose))
 
-  (plan, fraction) = move_arm.compute_cartesian_path(
+  (plan, fraction) = move_group.compute_cartesian_path(
                                waypoints,   # waypoints to follow
                                0.01,        # end effector follow step (meters)
                                0.0)         # jump threshold
+
   return plan, fraction
 
-def grind(move_arm, move_limbs, args):
+def grind(move_arm, move_limbs, move_grinder, args):
 
   x_start = args[1]
-  y_start = args[2]
+  y_start = args[2] 
```

Comment:
Small style comment: multi-assignment is a compact alternative: x_start, y_start = args[1], args[2]